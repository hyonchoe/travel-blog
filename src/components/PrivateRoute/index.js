/**
 * PrivateRoute component that checks for authentication,
 * otherwise redirects to main page.
 */

import React from 'react'
import PropTypes from 'prop-types'
import { Route, Redirect } from 'react-router-dom'
import { useAuth0 } from '@auth0/auth0-react'

const PrivateRoute = ({ component: Component, ...rest }) => {
    PrivateRoute.propTypes = {
        component: PropTypes.elementType.isRequired
    }
    
    const { isAuthenticated } = useAuth0()

    return (
        <Route
            {...rest}
            render = {(props) => (
                isAuthenticated ? 
                    <Component {...props} {...rest} />
                    : <Redirect to='/' />
            )} />
    )
}

export default PrivateRoute