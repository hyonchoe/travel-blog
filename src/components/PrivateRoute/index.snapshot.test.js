/**
 * Snapshot tests for PrivateRoute component.
 * 
 * Run by running 'npm test' in command line.
 */

import React from 'react'
import { shallow } from 'enzyme'
import PrivateRoute from './'

describe('PrivateRoute', () => {
    const DummyComponent = () => {
        return (
            <div className='dummy'></div>
        )
    }
    
    it('matches snapshot for my trips', () => {
        const handleEditTrip = jest.fn()
        const wrapper = shallow(
            <PrivateRoute 
                path='/myTrips'
                component={DummyComponent}
                showMyTrips={true}
                editTrip={handleEditTrip} />
        )
        expect(wrapper).toMatchSnapshot()
    })

    it('matches snapshot for add trips', () => {
        const clearEditTrip = jest.fn()
        const wrapper = shallow(
            <PrivateRoute
                path='/addTrip'
                component={DummyComponent}
                tripToEdit={null}
                clearEditTrip={clearEditTrip} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})