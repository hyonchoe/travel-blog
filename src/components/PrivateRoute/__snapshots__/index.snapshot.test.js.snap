// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute matches snapshot for add trips 1`] = `
<Route
  clearEditTrip={[MockFunction]}
  path="/addTrip"
  render={[Function]}
  tripToEdit={null}
/>
`;

exports[`PrivateRoute matches snapshot for my trips 1`] = `
<Route
  editTrip={[MockFunction]}
  path="/myTrips"
  render={[Function]}
  showMyTrips={true}
/>
`;
